code/modules/dwarven/dwarven_structures.dm

```diff
@@ -0,0 +1,428 @@
+	if(!H.has_language(/datum/language/dwarven))
+		to_chat(H, "<span class='notice'>You don't understand how to operate this ancient machinery!</span>")
+		return
+	if(isbodypart(I))
+		to_chat(H, "<span class='notice'>The [I.name] turns to ash as you impale it on the bone. Infuser brightly flashes and blood pool swells.</span>")
+		charge_amount++
+		qdel(I)
+		return
+	. = ..()
+
+/obj/structure/destructible/dwarven/blood_pool/attack_hand(mob/user)
+	if(!istype(user,/mob/living/carbon/human))
+		return
+	var/mob/living/carbon/human/H = user
+	if(!H.has_language(/datum/language/dwarven))
+		to_chat(H, "<span class='notice'>You don't understand the instructions written in that ancient tongue</span>")
+		return
+
+	if(charge_amount >= 30)
+		to_chat(H, "<span class='notice'>You hear a faint whisper telling you to choose wiselty </span>")
+		var/choice
+		choice = alert(user,"You study the schematics etched into the stone...",,"Blitz rune","Air rune","Earth rune")
+		switch(choice)
+			if("Blitz rune")
+				new /obj/item/dwarven/rune_stone/blitz(get_turf(src))
+			if("Air rune")
+				new /obj/item/dwarven/rune_stone/air(get_turf(src))
+			if("Earth rune")
+				new /obj/item/dwarven/rune_stone/earth(get_turf(src))
+		to_chat(H, "<span class='notice'>You hear a whisper telling you you have chosen wisely.</span>")
+		charge_amount -= 20
+		return
+
+	. = ..()
+
+/obj/structure/destructible/dwarven/mythril_press
+	name = "Ancient Alloy Press"
+	icon_state = "mythril_press"
+	custom_materials  = list(/datum/material/silver = 20000)
+
+/obj/structure/destructible/dwarven/mythril_press/Initialize()
+	AddComponent(/datum/component/material_container,list(/datum/material/gold,/datum/material/silver,/datum/material/titanium,/datum/material/dwarven,), 20000, TRUE, /obj/item/stack/sheet/mineral)
+	. = ..()
+
+/obj/structure/destructible/dwarven/mythril_press/attack_hand(mob/user)
+	var/mob/living/carbon/human/H = user
+	if(!H.has_language(/datum/language/dwarven))
+		to_chat(H, "<span class='notice'>You don't understand how to operate this ancient machinery!</span>")
+		return
+	press_mythril(user)
+	. = ..()
+
+/obj/structure/destructible/dwarven/mythril_press/proc/press_mythril(mob/user)
+	var/datum/component/material_container/materials = GetComponent(/datum/component/material_container)
+	if(materials.has_materials(list(/datum/material/gold= 1000, /datum/material/silver = 1000, /datum/material/titanium = 1000)))
+		materials.use_materials(list(/datum/material/gold= 1000, /datum/material/silver = 1000, /datum/material/titanium = 1000))
+		materials.insert_amount_mat(materials.sheet2amount(1),/datum/material/dwarven)
+		materials.retrieve_all(get_turf(src))
+		to_chat(user, "<span class='notice'>You hear a loud crank as materials are compressed into dwarven alloy!</span>")
+		return
+	materials.retrieve_all(get_turf(src))
+	to_chat(user, "<span class='notice'>The machine makes a loud crank sound, but no alloy falls out!</span>")
```

Comment:
this should return early instead of the huge block above, small block before big block